core/index/index_writer.cpp

```diff
@@ -232,12 +232,12 @@ bool RemoveFromImportedSegment(DocumentMask& deleted_docs,
     return false;
   }
 
-  auto prepared = query.filter->prepare(reader);
+  auto prepared = query.filter->prepare({.index = reader});
```

Comment:
We can pass here Transactions to account memory used during prepare/execute of remove filters.